Cooookiez/Programowanie-obiektowe
# Test 0

**Description**

This PR introduces a Google Test fremework into the project. 

**Does this PR introduce a breaking change?**

_Yes._ You have to install GoogleTest on your machine. 

_Caveat:_ Most modern installations of GTest require that it be downloaded as source code, compiled and deployed to the ultimate location. 

This link describes how to do this in Ubuntu: https://www.eriksmistad.no/getting-started-with-google-test-on-ubuntu/ assuming that you can `sudo` (that is, that you can act as the system's administrator). 

For this reason installing a GTest package, e.g. `> sudo apt-get install libgtest-dev`, will only install the source code of the library. You need to follow the instructions from the URL to compile (`cmake`, `make`) and deploy (`cp`) the library.  


**How to test this PR** 

- Switch to the PR's branch (`git checkout test_0`)
- Compile the project, either via QtCreator or other IDE, or manually
To compile the project manually, 


1.  switch to your build directory, e.g. `cd ../build-Pierwszy-Desktop-Default/` 
1.  run `cmake .` or `cmake ../Pierwszy`
1.  run `make`

-  Check that you have two new executables, `Pierwszy` and `test1`

```
> ls -ltr
...
-rwx---r-x 1 zkoza users 13501 Oct 16 18:40 Pierwszy
-rwx---r-x 1 zkoza users 13578 Oct 16 18:40 test1

```

- Run both executables. The test passes if none of the programs crashes. 

**When to merge** 

Merge when  all reviewers approved this PR. Wait for the classes to see yourself and let others see how to do this and what it changes to the project. 

**Other issues**

None

**Change log**

1. adds a new target, `test1`
1. adds a dependency on Google Test
